libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -50,22 +50,23 @@ public BotFrameworkAdapter(ICredentialProvider credentialProvider, RetryPolicy c
                 throw new ArgumentNullException(nameof(reference));
 
             if (callback == null)
-                throw new ArgumentNullException(nameof(callback)); 
+                throw new ArgumentNullException(nameof(callback));
 
-            var context = new TurnContext(this, reference.GetPostToBotMessage());
-
-            // Hand craft Claims Identity.
-            var claimsIdentity = new ClaimsIdentity(new List<Claim>
+            using (var context = new TurnContext(this, reference.GetPostToBotMessage()))
```

Comment:
> using (var context = new TurnContext(this, reference.GetPostToBotMessage())) [](start = 11, length = 77)

hmmm..IDisposable is a very C# pattern.  how does this translate to node and other langs?

Reply:
Yes, this is one of those .NET specific things. I would say you just don't do offer it in those languages if it doesn't make sense. 

An alternative, for something like TypeScript, would be to take another, option parameter when registering the service that is the call back you want invoked when the turn is ending.